### DISCUSSION TITLE
Uploading videos

### DISCUSSION THREAD

**Discussion by oxinabox** [pretalx/pretalx#1144] [Ideas] [answered]
For Juliacon, we have been online this year and last.
and we got everyone to upload videos in advance.

We briefly tried to use Pretalx's "upload resource".
But got an error about the file being too big.
Which makes sense.

We ended up using our own more janky solution that uploaded into a AWS S3 bucket.

But a solution that is integrated into Pretalx would be better because it would use the same authorization.
And would be able to do things like let any of the listed speakers upload, and it would let them check if the other speaker had uploaded etc.

**Comment by vmx** [comment]
As I was helping with an online conference that was also using pretalx, it might be interesting for people reading this, to see how other janky solutions looked like. We used [Seafile](https://www.seafile.com) for storage. I've written an [in-depth blog post](https://vmx.cx/cgi-bin/blog/index.cgi/video-uploads-for-an-online-conference%3A2021-06-12%3Aen%2Cconference%2Cgeo) on the whole process, the source code can be found at https://github.com/vmx/conference-tools/.

**Comment by rixx** [comment, marked as answer]
Integrating video storage into pretalx sounds good at the surface, but it's much more complicated than it looks. For example, good video providers transcode uploaded videos, and provide several download sizes/methods and a good player that goes beyond the standard HTML player. This goes way beyond what pretalx currently does, and what I know how to do. It would basically need another commited contributor who takes care of the video system. We currently don't even see outside contributors for much smaller and more solvable problems.

Additionally, speaking as a hoster, offering free and unlimited video space is wildly unrealistic, especially while we're also committed (as we currently are) to never removing old conferences, even after the payment period ends. Video files are large, especially if you retain multiple versions, and having guaranteed-to-grow storage gets expensive.

I don't see a solution to any of these problems. If anybody does have a solution, I'm more than happy to hear it.

That said, pretalx already supports the embedding of videos on talk pages via plugins – and as I understand it, that is the most likely use case? There currently exist plugins for youtube and media.ccc.de, and adding more providers (like for example mux.com or vimeo) wouldn't be a huge issue.

**Reply by vmx** [reply]
I agree with @rixx here. Though there might be things that could be added to pretalx, that could improve dealing with external resources. For example being able to add a link personalized upload link that only the speakers can see, or other customized fields fields to show that e.g. their video was received successfully (I should really open an issue with more details about this :)

**Reply by rixx** [reply]
I could see all of these, and also maybe a "video link" question type that would automatically embed the answer when reviewers look at it – if that sounds like something that would solve (or reduce) the issue for you, please open a separate issue!

**Reply by oxinabox** [reply]
The feature I want is only *upload* videos, no serve then, not transcodethem, not even host them.
The solution that would be ideal for us at JuliaCon would be if we could attach a AWS S3 bucket, (I guess others would want a dropbox or google drive etc), and just let people upload files into it, and download back again to check them.
As blackbox binary blobs.
We don't even want to expose the ability to download them to anyone other than the speaker /
We would then access the same bucket directly and upload it to youtube etc. We have our own system for that.

I agree, that hosting and serving videos would be a huge project.
Hooking up to some auth system would be simpler (but annoying to support multiple providers, but still simpler)

**Reply by rixx** [reply]
For blackbox uploads, I'd be willing to add the necessary hooks for a plugin (eg pretalx-dropbox-questions, pretalx-aws-questions) if anybody wants to contribute or commission it. I imagine it's a rather limited use case, but it would be nice to have in pretalx regardless.

**Reply by oxinabox** [reply]
@logankilpatrick can you get JuliaCon committee to talk about commissioning this if we are going to be online again next year?
(or partially online etc)

**Reply by toshywoshy** [reply]
Well, @rixx has a pretalx-vimeo plugin and I just made a pretalx-youtube plugin, I am sure we can add other external video embed / link plugins if needed. I will try to make a pretalx-video-s3 or a pretalx-video-nextcloud version, as I might have some use cases for those in the future.
